sretoolbox/container/image.py

```diff
@@ -174,28 +232,17 @@ def _get_manifest(self):
         # NOTE(efried): At least for quay, this returns schemaVersion 1 by tag
         # and 2 by digest.
         url += f'/{self.image}/manifests/{reference}'
-        if self.response_cache is None:
-            return self._request_get(url)
 
-        try:
-            rsp = self._request_get(url, requests.head)
-            return self.response_cache[rsp.headers['Docker-Content-Digest']]
-        except KeyError:
-            rsp = self._request_get(url)
-            if self._should_cache(rsp):
-                self.response_cache[rsp.headers['Docker-Content-Digest']] = rsp
-            return rsp
-
-    @classmethod
-    def _should_cache(cls, response):
-        try:
-            return (
-                'Docker-Content-Digest' in response.headers
-                and int(response.headers['content-length']) <
-                cls.MAX_CACHE_ITEM_SIZE
-            )
-        except (KeyError, ValueError):
-            return False
+        if self.response_cache is None or not self._can_response_be_cached():
+            return self._do_request(url)
+
+        key = self._get_cache_key(url)
+        if key in self.response_cache:
+            self._handle_cache_response(url)
```

Comment:
nits to improve readability and maintenance:
- inside the `else` you're populating the response_cache... for consistency, you should populate it here as well.
- the helpers don't have to manipulate the cache at all, you can just pass the cached_response

I mean:
```
        if key in self.response_cache:
            self.response_cache[key] = self._handle_cache_response(url, cached_response)
        else:
            self.response_cache[key] = self._do_request(url)
```

Reply:
Yeah, I also thought about this. I agree that the implementation is not pretty, I just wanted to avoid to assignment in case it is not necessary (when you don't need to update the cache), but I agree that it will look much more understandable this way and most likely the optimization is not necessary. I will change it.

Reply:
Refactored as suggested in 21859d8